rio_tiler_pds/landsat/aws/landsat_collection2.py

```diff
@@ -33,6 +34,8 @@
 from ... import get_object
 from ..utils import sceneid_parser
 
+s3_client = boto3.client("s3")
```

Comment:
🤔 I don't think you have to create a client if it's only to catch the error 

quick search on github https://github.com/Pkanjan37/Lightweight_big_data_framework_serverless/blob/f7edeb12c708bf66ba7175778381d6694fa649fd/pywren/storage/s3_backend.py#L79-L83

Reply:
Yeah I was looking at https://stackoverflow.com/questions/42975609/how-to-capture-botocores-nosuchkey-exception, and catching the exact error seems nicer to me, but I agree it's annoying to create a new client